op-e2e/e2eutils/addresses.go

```diff
@@ -0,0 +1,65 @@
+package e2eutils
+
+import (
+	"bytes"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/crossdomain"
+)
+
+func collectAllocAddrs(alloc core.GenesisAlloc) []common.Address {
+	var out []common.Address
+	for addr := range alloc {
+		out = append(out, addr)
+	}
+	// make output deterministic
+	sort.Slice(out, func(i, j int) bool {
+		return bytes.Compare(out[i][:], out[j][:]) < 0
+	})
+	return out
+}
+
+// CollectAddresses constructs a lists of addresses that may be used as fuzzing corpora
+// or random address selection.
+func CollectAddresses(sd *SetupData, dp *DeployParams) (out []common.Address) {
+	// This should be seeded with:
+	//  - reserve 0 for selecting nil (contract creation)
+	out = append(out, common.Address{})
```

Comment:
why put in the empty address twice?

Reply:
I'm thinking we reserve index 0 to be "nil", to handle the contract creation case (no destination address), while the other is really zero.

Reply:
ok. Does this imply that this is going to be migrated to `[]*common.Address` as the return type.

Reply:
Just `[]common.Address`, no pointer. Only when used as destination address it's useful to make "none" a part of the options. Having a duplicate zeroed as fallback is fine. Or we can remove the `nil` case altogether, and fuzz that separately.

Reply:
so calling code that selects an address from this list would turn the zero address into a nil pointer if the index it selected was 0?

Reply:
that was the idea yes. Not perfect, but that would make it easy to select the nil case as part of all possible addresses when creating a fuzzer.

Reply:
Ok. That's a little non intuitive, but it won't cause problems. It's a good reminder for when fuzzing gets implemented that if a function takes a pointer, we should be passing `nil`.